examples/examples/logger.rs

```diff
@@ -28,19 +28,19 @@ use std::net::SocketAddr;
 use std::time::Instant;
 
 use jsonrpsee::core::client::ClientT;
-use jsonrpsee::core::logger::{self, Headers, MethodKind, Params};
-use jsonrpsee::rpc_params;
+use jsonrpsee::server::logger::{self, HttpRequest, MethodKind, Params};
+use jsonrpsee::server::ServerBuilder;
 use jsonrpsee::ws_client::WsClientBuilder;
-use jsonrpsee::ws_server::{RpcModule, WsServerBuilder};
+use jsonrpsee::{rpc_params, RpcModule};
 
 #[derive(Clone)]
 struct Timings;
 
-impl logger::WsLogger for Timings {
+impl logger::Logger for Timings {
 	type Instant = Instant;
 
-	fn on_connect(&self, remote_addr: SocketAddr, headers: &Headers) {
-		println!("[Logger::on_connect] remote_addr {}, headers: {:?}", remote_addr, headers);
+	fn on_connect(&self, remote_addr: Option<SocketAddr>, request: &HttpRequest) {
```

Comment:
When wouldn't we have a `SocketAddr`?

Reply:
The issue is that `socket.peer_addr -> Result<SocketAddr, Error>` and I don't really know when this can fail assuming that it won't but yeah.... awkward API for sure.

Reply:
Ah ok, that's fair!